src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt

```diff
@@ -468,6 +477,23 @@ class VirtualMachine(
                         Pair(listOf(stackInit), stackInit)
                     }
                     linearRepresentation.addAll(instructions)
+
+                    val newObjectStyle = stackValue.style.animate ?: stackValue.style
+                    node.initialValue.map { executeExpression(it) }.forEach {
+                        stackValue.stack.push(it)
+                        val rectangle = Rectangle(
+                            variableNameGenerator.generateNameFromPrefix("rectangle"),
+                            it.value.toString(),
+                            initStructureIdent,
+                            color = newObjectStyle.borderColor,
+                            textColor = newObjectStyle.textColor
+                        )
+                        linearRepresentation.add(NewMObject(
+                            rectangle,
+                            codeTextVariable
+                        ))
+                        linearRepresentation.add(StackPushObject(rectangle, initStructureIdent))
+                    }
```

Comment:
480 - 496 is the new code, I moved the method calls into separate functions too above as they were getting really big